dynamic/object/metadata.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2018 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package object
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// HasFinalizer returns true if obj has the named finalizer.
+func HasFinalizer(obj metav1.Object, name string) bool {
+	for _, item := range obj.GetFinalizers() {
+		if item == name {
+			return true
+		}
+	}
+	return false
+}
+
+// AddFinalizer adds the named finalizer to obj, if it isn't already present.
+func AddFinalizer(obj metav1.Object, name string) {
```

Comment:
This function can return a `bool` value, in which `true` indicates the finalizer has been set, and `false` if the finalizer already exists (no-op).

(But I think you have reason for not doing so - most probably due to coding style.)

Reply:
We can discuss this in the thread above.